hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/integration/CompactFormatIntegrationTest.java

```diff
@@ -188,4 +191,27 @@ public Object process(Map.Entry<Integer, GenericRecord> entry) {
             return null;
         }
     }
+
+    @Test
+    public void testMapListener() {
+        IMap<Integer, EmployeeDTO> map = instance1.getMap("test");
+        CountDownLatch latch = new CountDownLatch(1);
+
+        class MapListener implements EntryAddedListener {
+            @Override
+            public void entryAdded(EntryEvent event) {
+                latch.countDown();
+            }
+        }
+        map.addEntryListener(new MapListener(), true);
+
+        // Put the entry from the other instance to not create a local
+        // registry in the actual instance. This will force it to
+        // go the cluster to fetch the schema.
+        IMap<Integer, EmployeeDTO> map2 = instance2.getMap("test");
+        EmployeeDTO employeeDTO = new EmployeeDTO(30, 102310312);
+        map2.put(1, employeeDTO);
+
+        assertOpenEventually(latch);
+    }
```

Comment:
This test is not really needed for the Java client. It was required in the Python client to make sure that we do schema fetching in the listener service non-blocking, but in Java, we have the blocking toObject